### PR TITLE
Resolve bugs

### PR DESCRIPTION
**Pull request by anandsinghparihar**
1) Passphrase recovery inputs need to be more responsive (Lower resolution)
2) Escape doesn't always work on modals - Wallet creation modal / password modal
3) Disable close modal while midway through wallet creation or restore
4) On the receive page, add a New Address modal instead of using a html dialogue

### CODE REVIEW

File: src/app/modals/modals.component.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export class ModalsComponent implements DoCheck, OnInit {
   syncPercentage: number = 0;
   syncString: string;
   closeOnEscape: boolean = true;
+  disableClose: object; // @TODO CREATE model for object
```
Comment (rynomster) [inline src/app/modals/modals.component.ts]: Don't use negative variable names, it should be 
enableClose 
Also not sure why it's an object?

File: src/app/wallet/receive/receive.component.html [hunk 2]
```diff
@@ -115,8 +115,26 @@
...
         </div>
       </div>
-      <button (click)="newAddress()">new {{ getAddressType() }} address</button>
+      <button (click)="openNewAddress()">new {{ getAddressType() }} address</button>
     </div>
 
   </div>
 </div>
+
+<div id="address-modal" [ngClass]="{'hide': openNewAddressModal == false }">
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.html]: Use `[class.hide]=` for single classes

File: src/app/wallet/receive/receive.component.scss [hunk 3]
```diff
@@ -187,4 +187,75 @@ button {
...
+  position: absolute;
+  z-index: 100000;
+  top: 0;
+  left: 0;
+  width: 100vw;
+  height: 100vh;
+  background-color: rgba(0, 0, 0, 0.3);
+}
+
+#address-modal.hide {
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.scss]: This is not proper scss

File: src/app/wallet/receive/receive.component.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
-import { Component, OnInit, HostListener, ElementRef, ViewChild } from '@angular/core';
-import { RPCService } from '../../core/rpc/rpc.service';
+import {Component, OnInit, HostListener, ElementRef, ViewChild} from '@angular/core';
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.ts]: Follow current standards in our codebase { .... }

File: src/app/wallet/receive/receive.component.ts [hunk 5]
```diff
@@ -112,10 +132,10 @@ export class ReceiveComponent implements OnInit {
   /* ---- UI Helper functions ---------------------------------------------- */
 
   /**
-    * Returns whether we're in search mode or not!
-    * The current table is showing limited results due to search.
-    * Mainly for hiding the "Unused address" & ease of use in other functions.
-    */
+   * Returns whether we're in search mode or not!
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.ts]: Standard comments :
```
  /**
    *
    *
```

File: src/app/wallet/receive/receive.component.ts [hunk 6]
```diff
@@ -171,21 +191,21 @@ export class ReceiveComponent implements OnInit {
...
   /**
-    * Used to get the addresses.
-    * TODO: Create interface Array<AddressInterface?>
-    */
+   * Used to get the addresses.
+   * TODO: Create interface Array<AddressInterface?>
+   */
   rpc_loadAddresses_success(response: Array<any>) {
     const pub = [],
-          priv = [];
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.ts]: Fix readability

File: src/app/wallet/receive/receive.component.ts [hunk 7]
```diff
@@ -206,7 +226,7 @@ export class ReceiveComponent implements OnInit {
       this.addresses.private = [];
     }
 
-    pub .forEach((val) => this.addAddress(val, 'public'));
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.ts]: Fix readability

File: src/app/wallet/wallet.module.ts [hunk 8]
```diff
@@ -21,6 +21,7 @@ import { BalanceComponent } from './balances/balance.component';
 import { HistoryComponent } from './history/history.component';
 
 import { AddressLookupComponent } from './addresslookup/addresslookup.component';
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
```
Comment (rynomster) [inline src/app/wallet/wallet.module.ts]: Follow codebase standards `{ .... }`

File: src/app/modals/createwallet/createwallet.component.html [hunk 9]
```diff
@@ -37,7 +37,7 @@
         <label>YOUR WALLET NAME</label>
       </div>
 
-      <div class="col-md-7">
```
Comment (rynomster) [inline src/app/modals/createwallet/createwallet.component.html]: This squishes the fields, it doesn't look good

File: src/app/modals/modals.component.ts (hunk 1 above)
Comment (anandsinghparihar) [inline src/app/modals/modals.component.ts]: Yes name changes, and I had define Object in service so we can add more property later in the same object.

File: src/app/modals/createwallet/createwallet.component.html [hunk 10]
```diff
@@ -37,7 +37,7 @@
...
+      <div class="col-md-6">
```
Comment (rynomster) [inline src/app/modals/createwallet/createwallet.component.html]: This is still squished

File: src/app/modals/createwallet/createwallet.component.html [hunk 11]
```diff
@@ -155,7 +155,7 @@
...
```
Comment (rynomster) [inline src/app/modals/createwallet/createwallet.component.html]: This is still squished

File: src/app/modals/createwallet/createwallet.component.html [hunk 12]
```diff
@@ -97,7 +97,7 @@
...
```
Comment (rynomster) [inline src/app/modals/createwallet/createwallet.component.html]: This is still squished

File: src/app/modals/modals.component.html [hunk 13]
```diff
@@ -3,7 +3,7 @@
   <div class="modal-dialog modal-sm">
     <div class="modal-content">
 
-      <button type="button" class="close pull-right" aria-label="Close" id="close" (click)="close()">
+      <button type="button" class="close pull-right" aria-label="Close" id="close" (click)="close()" [disabled]="enableClose.status">
```
Comment (rynomster) [inline src/app/modals/modals.component.html]: Should rather hide the button at this stage

File: src/app/modals/modals.component.ts [hunk 14]
```diff
@@ -63,8 +64,9 @@ export class ModalsComponent implements DoCheck, OnInit {
   }
 
   ngOnInit() {
+    this.enableClose = this._modalService.enableClose;
     document.onkeydown = (event: any) => {
-      if (this.closeOnEscape
+      if (this.closeOnEscape && !this.enableClose['status']
```
Comment (rynomster) [inline src/app/modals/modals.component.ts]: This logic should be if enableClose

File: src/app/modals/modals.service.ts [hunk 15]
```diff
@@ -16,6 +16,11 @@ export class ModalsService {
   public modal: any = null;
   private message: Subject<any> = new Subject<any>();
   private progress: Subject<Number> = new Subject<Number>();
+  public closeButton: Subject<any> = new Subject<any>();
+  // @TODO CREATE model for object
+  public enableClose: object = {
```
Comment (rynomster) [inline src/app/modals/modals.service.ts]: This should default to true

File: src/app/wallet/receive/receive.component.spec.ts [hunk 16]
```diff
@@ -158,4 +159,32 @@ describe('ReceiveComponent', () => {
     });
     */
 
+    describe('#newAddress()', function () {
+      it('is define', function () {
+        expect(component.newAddress).toBeDefined();
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.spec.ts]: This test is pointless

File: src/app/wallet/receive/receive.component.spec.ts [hunk 17]
```diff
@@ -158,4 +159,32 @@ describe('ReceiveComponent', () => {
...
+    });
+
+    describe('#openNewAddress()', function () {
+      it('is define', function () {
+        expect(component.openNewAddress).toBeDefined();
+      });
+
+      it('should be open new address modal', function () {
+        component.openNewAddress();
+        expect(component.openNewAddressModal).toEqual(true);
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.spec.ts]: This test is pointless

File: src/app/wallet/receive/receive.component.spec.ts [hunk 18]
```diff
@@ -158,4 +159,32 @@ describe('ReceiveComponent', () => {
...
+      })
+    });
+
+  describe('#closeNewAddress()', function () {
+    it('is define', function () {
+      expect(component.closeNewAddress).toBeDefined();
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.spec.ts]: This test is pointless

**Comment by rynomster** [comment]
This still doesn't solve the recovery phrase inputs

File: src/app/modals/modals.component.html [hunk 19]
```diff
@@ -3,7 +3,7 @@
   <div class="modal-dialog modal-sm">
     <div class="modal-content">
 
-      <button type="button" class="close pull-right" aria-label="Close" id="close" (click)="close()">
+      <button type="button" class="close pull-right" aria-label="Close" id="close" (click)="close()" [disabled]="!enableClose.status">
```
Comment (rynomster) [inline src/app/modals/modals.component.html]: As previously discussed.. This should be hidden

File: src/app/modals/modals.service.ts [hunk 20]
```diff
@@ -16,6 +16,11 @@ export class ModalsService {
   public modal: any = null;
   private message: Subject<any> = new Subject<any>();
   private progress: Subject<Number> = new Subject<Number>();
+  public closeButton: Subject<any> = new Subject<any>();
```
Comment (rynomster) [inline src/app/modals/modals.service.ts]: I don't see where this gets used?

File: src/app/wallet/receive/receive.component.scss [hunk 21]
```diff
@@ -187,4 +190,77 @@ button {
...
+}
+
+.modal-label {
+  text-transform: uppercase;
+  font-weight: bold;
+  font-size: 0.7em;
+  margin-bottom: 10px;
+}
+
+.good {
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.scss]: stick to .valid .invalid ?

Comment (rynomster) [reply src/app/wallet/receive/receive.component.scss]: we also have other validation classes

File: src/app/wallet/receive/receive.component.ts [hunk 22]
```diff
@@ -223,17 +243,17 @@ export class ReceiveComponent implements OnInit {
...
+    */
   addAddress(response: any, type: string) {
     const tempAddress = {
       id: 0,
       label: 'Empty label',
       address: 'Empty address',
       balance: 0,
       readable: ['Empty']
-    }
+    };
```
Comment (rynomster) [inline src/app/wallet/receive/receive.component.ts]: We don't have semi colons after typescript functions

File: src/app/wallet/receive/receive.component.scss (hunk 21 above)
Comment (anandsinghparihar) [reply src/app/wallet/receive/receive.component.scss]: Can you explain this comment?

File: src/app/modals/createwallet/createwallet.component.html (hunk 12 above)
Comment (rynomster) [reply src/app/modals/createwallet/createwallet.component.html]: this

File: src/app/modals/createwallet/createwallet.component.html (hunk 10 above)
Comment (rynomster) [reply src/app/modals/createwallet/createwallet.component.html]: this

File: src/app/modals/createwallet/createwallet.component.html (hunk 9 above)
Comment (rynomster) [reply src/app/modals/createwallet/createwallet.component.html]: this

**Pull request merged**